picatrix/magics/common_test.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the picatrix framework."""
+from typing import Optional
+from typing import Text
+
+from picatrix.lib import framework
+from picatrix.lib import manager
+from picatrix.magics import common
+
+
+def my_very_own_test_magic(data: Text, stuff: Optional[int] = 20) -> Text:
+  """This is a magic that is used for testing.
+
+    Args:
+        data (str): This is a string.
+        stuff (int): And this is a number.
+
+    Returns:
+        str: A string that combines the two parameters together.
+  """
+  return f'{data.strip()} - {stuff}'
+
+
+def test_registration():
+  """Test the magic decorator."""
+  manager.MagicManager.clear_magics()
+  _ = framework.picatrix_magic(my_very_own_test_magic)
+  magic = framework.picatrix_magic(common.picatrixmagics)
+
+  df = magic(line='')
+  assert df.shape[0] == 2
```

Comment:
Why not

```
assert df.columns == ['name', 'desc', ...]
```

Reply:
As I mentioned before, perhaps len(df) is clearer.

Reply:
ok, so 

`df.shape[0]` is the number of lines

I prefer shape rather than len, been discouraged in many places